### PR TITLE
coord: add safe mode concept

### PR DESCRIPTION
**Pull request by benesch**
Add the concept of "safe mode" to Materialize, which can be controlled
via the undocumented "--safe" command-line option. It is intended for
use in the cloud product, where features that access the machine running
Materialize need to be disabled.

In particular, it disables:

  * File sources and sinks, because they allow reading/writing files on
    the local machine.

  * Avro OCF sources and sinks, by the same logic.

  * Kafka sources and sinks that use Kerberos authentication, because
    librdkafka's implementation permits specification of an arbitrary
    shell command (kinit_cmd) to run periodically. Kerberos auth is not
    a good fit for the cloud product, because it requires quite a bit of
    control over the network on which the machines run, so it's easiest
    just to disable Kerberos support entirely for now.

### CODE REVIEW

File: src/coord/src/coord.rs [hunk 1]
```diff
@@ -774,6 +776,14 @@ impl Coordinator {
                             },
                         }
 
+                        if let Err(e) = check_statement_safety(&stmt) {
```
Comment (umanwizard) [inline src/coord/src/coord.rs:779]: Shouldn't this check be conditional on safe mode being enabled?

Comment (benesch) [reply src/coord/src/coord.rs]: D'oh, yes.

**Review by umanwizard — APPROVED** [review]

**Comment by benesch** [comment]
Added some tests, and actually got the kerberos auth bit right. Mind taking one more look?

**Comment by benesch** [comment]
I'm actually gonna merge to unblock #6352, but happy to address feedback post merge.

**Pull request merged**